File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/PlayerAdapter.java
package com.garpr.android.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;

import com.garpr.android.R;
import com.garpr.android.models.LiteTournament;
import com.garpr.android.models.Match;
import com.garpr.android.models.Rating;

import java.util.HashMap;
import java.util.Map;

public class PlayerAdapter extends BaseMultiAdapter {

    private static final Map<Class, Integer> LAYOUT_KEY_MAP;


    static {
        LAYOUT_KEY_MAP = new HashMap<>(4);
        LAYOUT_KEY_MAP.put(LiteTournament.class, R.layout.divider_tournament);
        LAYOUT_KEY_MAP.put(Match.class, R.layout.item_match);
        LAYOUT_KEY_MAP.put(Rating.class, R.layout.item_rating);
        LAYOUT_KEY_MAP.put(String.class, R.layout.item_string);
    }

    public PlayerAdapter(@NonNull final Context context) {
        super(context, LAYOUT_KEY_MAP);
    }

    public PlayerAdapter(@NonNull final LayoutInflater layoutInflater) {
        super(layoutInflater, LAYOUT_KEY_MAP);
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/PlayerAdapter.kt
package com.garpr.android.adapters

import android.content.Context
import com.garpr.android.R
import com.garpr.android.data.models.FullPlayer
import com.garpr.android.data.models.LiteTournament
import com.garpr.android.data.models.Match

class PlayerAdapter(context: Context) : BaseMultiAdapter(context, LAYOUT_KEY_MAP) {

    companion object {
        private val LAYOUT_KEY_MAP = mapOf<Class<*>, Int>(
                FullPlayer::class.java to R.layout.item_player_profile,
                LiteTournament::class.java to R.layout.divider_tournament,
                Match::class.java to R.layout.item_match,
                String::class.java to R.layout.item_string)
    }

    init {
        setHasStableIds(true)
    }

    override fun getItemId(position: Int): Long {
        return getItem(position).hashCode().toLong()
    }

}
